tests/js/client/shell/shell-api-cusror.js

```diff
@@ -0,0 +1,1209 @@
+
+'use strict';
+
+const jsunity = require("jsunity");
+
+const sleep = require('internal').sleep;
+let api = "/_api/cursor";
+let reId = /^\d+$/;
+const forceJson = internal.options().hasOwnProperty('server.force-json') && internal.options()['server.force-json'];
+const contentType = forceJson ? "application/json; charset=utf-8" :  "application/x-velocypack";
+
+////////////////////////////////////////////////////////////////////////////////;
+// error handling;
+////////////////////////////////////////////////////////////////////////////////;
+
+function dealing_with_cursorsSuite_error_handlingSuite () {
+  return {
+    test_returns_an_error_if_body_is_missing: function() {
+      let cmd = api;
+      let doc = arango.POST_RAW(cmd, "");
+
+      assertEqual(doc.code, 400);
+      assertEqual(doc.headers['content-type'], contentType);
+      assertTrue(doc.parsedBody['error']);
+      assertEqual(doc.parsedBody['code'], 400);
+      assertEqual(doc.parsedBody['errorNum'], 1502);
+    },
+
+    test_returns_an_error_if_query_attribute_is_missing: function() {
+      let cmd = api;
+      let body = "{ }";
+      let doc = arango.POST_RAW(cmd, body);
+
+      assertEqual(doc.code, 400);
+      assertEqual(doc.headers['content-type'], contentType);
+      assertTrue(doc.parsedBody['error']);
+      assertEqual(doc.parsedBody['code'], 400);
+      assertEqual(doc.parsedBody['errorNum'], 600);
+    },
+
+    test_returns_an_error_if_query_is_null: function() {
+      let cmd = api;
+      let body = "{ \"query\" : null }";
+      let doc = arango.POST_RAW(cmd, body);
+
+      assertEqual(doc.code, 400);
+      assertEqual(doc.headers['content-type'], contentType);
+      assertTrue(doc.parsedBody['error']);
+      assertEqual(doc.parsedBody['code'], 400);
+      assertEqual(doc.parsedBody['errorNum'], 1502);
+    },
+
+    test_returns_an_error_if_query_string_is_empty: function() {
+      let cmd = api;
+      let body = "{ \"query\" : \"\" }";
+      let doc = arango.POST_RAW(cmd, body);
+
+      assertEqual(doc.code, 400);
+      assertEqual(doc.headers['content-type'], contentType);
+      assertTrue(doc.parsedBody['error']);
+      assertEqual(doc.parsedBody['code'], 400);
+      assertEqual(doc.parsedBody['errorNum'], 1502);
+    },
+
+    test_returns_an_error_if_query_string_is_just_whitespace: function() {
+      let cmd = api;
+      let body = "{ \"query\" : \"    \" }";
+      let doc = arango.POST_RAW(cmd, body);
+
+      assertEqual(doc.code, 400);
+      assertEqual(doc.headers['content-type'], contentType);
+      assertTrue(doc.parsedBody['error']);
+      assertEqual(doc.parsedBody['code'], 400);
+      assertEqual(doc.parsedBody['errorNum'], 1501);
+    },
+
+    test_returns_an_error_if_collection_is_unknown: function() {
+      let cmd = api;
+      let body = { "query" : "FOR u IN unknowncollection LIMIT 2 RETURN u.n", "count" : true, "bindVars" : {}, "batchSize" : 2 };
+      let doc = arango.POST_RAW(cmd, body);
```

Comment:
This will not work. You will need to turn either `body` into a string, or stringify it using `JSON.stringify(body)` when sending it.